src/cmd/ksh93/sh/path.c

```diff
@@ -1507,6 +1507,10 @@ static int path_chkpaths(Shell_t *shp,Pathcomp_t *first, Pathcomp_t* old,Pathcom
 	if((fd=open(stakptr(offset),O_RDONLY))>=0)
 	{
 		fstat(fd,&statb);
+		if (S_ISDIR(statb.st_mode)) {
```

Comment:
What if it's a FIFO or a device file? I think it's safer to reject everything that isn't a regular file or a symlink to a regular file.

Reply:
Yep, checking explicitly to confirm if it is a regular file makes sense.

Symlink could be a problem if it's linking to a directory I guess.

Reply:
I'll test and send an update...

Reply:
Thanks. I could be wrong but I'm pretty sure S_ISREG resolves symlinks (like test -f) so it tests if it's either a regular file or a symlink to a regular file, which is probably just what we want here.